tailnet/configmaps.go

```diff
@@ -261,6 +276,26 @@ func (c *configMaps) setAddresses(ips []netip.Prefix) {
 	c.Broadcast()
 }
 
+func (c *configMaps) addHosts(hosts map[dnsname.FQDN][]netip.Addr) {
+	c.L.Lock()
+	defer c.L.Unlock()
+	for name, addrs := range hosts {
+		c.hosts[name] = append(c.hosts[name], addrs...)
```

Comment:
I think it's going to be hard to use this if we define it this way: that addresses append, rather than replace the list.  Think about the WorkspaceUpdates RPC.  Some of the agents will have changed their Agent ID due to rebuild.  We want DNS to stop using the old AgentID-derived IP address and immediately start returning the new one, but this will just add the new address in addition to the old.

We're also going to want a "set" functionality, for when we reconnect and get the initial full dump of workspaces and agents. Blow away all bindings we used to have and just use the new set.

Reply:
That makes sense. We don't need to support multiple IPs per hostname right? If so the slice can just always be one IP.

Reply:
Actually, I'll just make the `addHosts` function set the provided slice instead of appending, or limiting it to a single IP.